huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -312,7 +312,9 @@ func getProductDetail(config *config.Config, d *schema.ResourceData, engineType
 		}
 		for _, v := range ps.Values {
 			for _, p := range v.Details {
-				if engineType == "kafka" {
+				// All informations of product for single instance type and the kafka engine type are stored in the
```

Comment:
the new API is:  https://dms.cn-north-4.myhuaweicloud.com/v2/kafka/products
the struct is the same for single and cluster

Reply:
The function has been split into two functions